File: ___build/dsa_problem_set/largest_no_formation_from_array/largest_no_formation_from_array.js
function largestNumber(nums) {
  const compare = (a, b) => {
    const sortA = String(a) + String(b);
    const sortB = String(b) + String(a);
    return sortB.localeCompare(sortA);
  };

  nums.sort(compare);

  const allZeroes = nums.every(num => num === 0);
  return allZeroes ? "0" : nums.join("");
}

const array = [10, 2];
const largestNo = largestNumber(array);

console.log(largestNo);

File: ___build/dsa_problem_set/largest_no_formation_from_array/largest_no_formation_from_array.java
class Main {
    public static String largestNumber(int[] nums) {
        Comparator<String> compare = (a, b) -> {
            String sortA = a + b;
            String sortB = b + a;
            return sortB.compareTo(sortA);
        }

        String[] newNums = new String[nums.length];
        for(int i = 0; i < nums.length; i++) {
            newNums[i] = String.valueOf(nums[i]);
        }

        Arrays.sort(newNums, compare);

        boolean allZeroes = true;
        for(int num : nums) {
            if(num != 0) {
                allZeroes = false;
                break;
            }
        }

        return allZeroes ? "0" : String.join("", newNums);
    }

    public static void main(String[] args) {
        int[] nums = {10, 2};
        System.out.println(largestNumber(nums));
    }
}